transformer_engine/pytorch/module/linear.py

```diff
@@ -797,6 +784,57 @@ def __init__(
         else:
             self.gemm_bias_unfused_add = False
 
+    def init_weight_tensor(self, weights: torch.Tensor) -> torch.Tensor:
```

Comment:
Should we move these to `base.py` since these are general functions that would be required when we expand this to the other modules?

Reply:
I think we can move them with the follow up PR.

Reply:
I moved them in this PR and probably need to tinker with it as I test other modules besides Linear in follow-up PRs, but in the meantime please take a look and let me know what you think about the approach.